### DISCUSSION TITLE
How does staking work in Saito ?

### DISCUSSION THREAD

**Discussion by SankaD** [SaitoTech/saito-rust-workspace#115] [General]
1. What is staking in Saito ?
2. Why is staking required? What attack vectors are solved from it ?
3. How does staking work in Saito ?

**Comment by trevelyan** [comment]
1. what is staking in Saito

When blocks are produced, a portion of the incoming fees (block reward) is put into a staking treasury. When transactions are rebroadcast, they receive a portion of this "staking treasury".

2. Why is staking required?

We cannot control the variance of mining / golden-ticket production (i.e. sometimes miners will produce 2 golden tickets in the space of a single block, and sometimes they will produce none). It is not a problem if the network produces two tickets in the space of a single block because only one can be selected for inclusion in the longest-chain. But it is a problem if they cannot find any golden tickets.

In order to understand why this is a problem, the questioner must be able to think through the following problems:

 - what happens if we do not pay out every block?
 - what happens if we collect the unpaid tokens and distribute them back as a block reward?
 - can we lower difficulty to guarantee 1 golden ticket per block? what does this imply for cost-of-attack?

Staking solves these problems by allowing us to use 1 golden ticket to pay out multiple blocks, making the network indifferent to the short-term variance in the pace of golden ticket production. 

3. How does staking work in Saito?

Please see existing documentation:

https://github.com/SaitoTech/saito-implementation-proposals/blob/main/proposals/001_simplified_staking.md

**Comment by arpee** [comment]
https://docs.google.com/presentation/d/1OB1YDy91zJ5Dy6cc2Xh2MPXgCeiNOS1dgxzct5MUM4E/edit#slide=id.g109e694a47c_0_194

This is a useful document on the role of staking.
